pkg/aws/policies.go

```diff
@@ -487,8 +487,8 @@ func (c *awsClient) FindRoleARNs(roleType string, version string) ([]string, err
 }
 
 // FIXME: refactor similar calls to use this instead
-func (c *awsClient) ValidateAccountRoleVersionCompatibility(
-	roleName string, roleType string, minVersion string) (bool, error) {
+func (c *awsClient) ValidateAccountRoleVersionCompatibility(roleName string, roleType string, minVersion string,
```

Comment:
Same as above, is it worth breaking up this func to be clearer of its intent? 

If not, can we at least provide a unit test for this :pray:

Reply:
Sure, let me break the function.
Inside the function, we have a call to `iamClient.ListRoleTags`, do we have a mock for the AWS client in the unit tests?

Reply:
Alternatively, maybe it will be possible to refactor the function in a way that won't make an API call to AWS.

Reply:
I agree, reduce the need for the API call, and we can easily unit test the validation logic